spec/bolt/cli_spec.rb

```diff
@@ -1,1120 +1,1204 @@
 # frozen_string_literal: true
 
 require 'spec_helper'
+
+require 'bolt/cli'
+
+require 'bolt_spec/env_var'
 require 'bolt_spec/files'
 require 'bolt_spec/project'
-require 'bolt_spec/task'
-require 'bolt/cli'
-require 'bolt/util'
-require 'concurrent/utility/processor_counter'
-require 'r10k/action/puppetfile/install'
-require 'yaml'
 
-describe "Bolt::CLI" do
+describe Bolt::CLI do
+  include BoltSpec::EnvVar
   include BoltSpec::Files
   include BoltSpec::Project
-  include BoltSpec::Task
-
-  let(:inventory) { Bolt::Inventory.empty }
-  let(:target) { inventory.get_target('foo') }
 
   before(:each) do
-    outputter = Bolt::Outputter::Human.new(false, false, false, false, StringIO.new)
-
-    allow_any_instance_of(Bolt::CLI).to receive(:outputter).and_return(outputter)
-    allow_any_instance_of(Bolt::CLI).to receive(:warn)
-
-    # Don't print error messages to the console
+    # Disable printing to the console.
     allow($stdout).to receive(:puts)
     allow($stderr).to receive(:puts)
 
-    # Don't allow tests to override the captured log config
+    # Don't allow the logger to be configured, it'll leak doubles all over the
+    # place.
     allow(Bolt::Logger).to receive(:configure)
 
-    Logging.logger[:root].level = :info
+    # Disable analytics screen view. It doesn't like doubles. :(
+    allow_any_instance_of(described_class).to receive(:submit_screen_view)
+
+    # Stub all the things.
+    allow(Bolt::Analytics).to receive(:build_client).and_return(analytics)
+    allow(Bolt::Application).to receive(:new).and_return(application)
+    allow(Bolt::Config).to receive(:from_project).and_return(config)
+    allow(Bolt::Executor).to receive(:new).and_return(executor)
+    allow(Bolt::Inventory).to receive(:from_config).and_return(inventory)
+    allow(Bolt::Outputter).to receive(:for_format).and_return(outputter)
+    allow(Bolt::PAL).to receive(:new).and_return(pal)
+    allow(Bolt::Plugin).to receive(:setup).and_return(plugin)
+    allow(Bolt::Project).to receive(:create_project).and_return(project)
+    allow(Bolt::Rerun).to receive(:new).and_return(rerun)
+
+    # Allow the stubbed outputter to yield from its spinner.
+    allow(outputter).to receive(:spin) { |&block| block.call }
```

Comment:
I'm not sure I understand? These tests stub the outputter, and this line is specifically just to allow it to execute the block, otherwise it just returns the stub.